proposals/NNNN-support-for-android-custom-routerservice.md

```diff
@@ -25,42 +25,42 @@ Currently, AndroidManifest.xml can indicate the application has custom RouterSer
 
 ## Proposed solution
 
 Suppose a custom RouterService is basically designed for OEM-specific head unit, and the head unit works only with that specific custom RouterService.
-The current logic in previous section does not work very well in this scenario, because it heavily depends on the timing when the app calls SdlBroadcastReceiver#queryForConnectedService.
+The current logic in the previous section does not work very well in this scenario, because it heavily depends on the timing of when the app calls SdlBroadcastReceiver.queryForConnectedService.
 To improve the connectivity of custom RouterService, the approach would be:
 
-1. Do not rely on user app to call queryForConnectedService; instead, SdlProxy should automatically checks to see the RouterService that connects with SDL Core whenever needed.
+1. Do not rely on the user app to call queryForConnectedService; instead, SdlProxy should automatically check to see if the RouterService connects with SDL Core whenever needed.
 
-2. Custom RouterService can not rely on other apps to be waken up, because of its priority order. Instead, it needs to start RouterService by himself.
+2. Custom RouterService cannot rely on other apps to be woken up, because of its priority order. Instead, it needs to start RouterService on its own.
 
-3. We can reuse the existing logic that verifies the trusted RouterService. If a RouterService is not trusted, SdlProxy can wake up the best possible RouterService in the same way as current proxy does. 
+3. We can reuse the existing logic that verifies the trusted RouterService. If a RouterService is not trusted, SdlProxy can wake up the best possible RouterService in the same way the current proxy does. 
 
-In approach #1, the important point is the location where we find the RouterService its primary transport connects with head unit. I think it is in RouterServiceValidator class.
+In approach #1, the important point is the location where the RouterService's primary transport connects with the head unit. It should be the RouterServiceValidator class.
 
 ### RouterServiceValidator should pay attention to currently connected RouterService.
 
-When would be the expected time to check if a RouterService connects with head unit or not? That would be right before the app binds to RouterService, and that should be done automatically without relying on user application.
-The proposed solution is to add new asynchronous method to RouterServiceValidator, and calls it right before TransportManager connects to the RouterService.
+The expected time to check if a RouterService connects with head unit would be right before the app binds with the RouterService, and that should be done automatically without relying on user application.
```

Comment:
It does seem a little long, would this work:

The expected time to check if a RouterService connects with head unit would be right before the app binds with the **RouterService - this** should be done automatically without relying on user application.

Reply:
Sounds good. Thanks.